src/librustdoc/html/format.rs

```diff
@@ -824,32 +745,16 @@ fn fmt_type(t: &clean::Type, f: &mut fmt::Formatter, use_absolute: bool,
                 _ => true,
             };
             if f.alternate() {
-                if is_not_debug {
-                    if should_show_cast {
-                        write!(f, "<{:#} as {:#}>::", self_type, trait_)?
-                    } else {
-                        write!(f, "{:#}::", self_type)?
-                    }
+                if should_show_cast {
```

Comment:
The `is_not_debug == false` case was:

```rust
if should_show_cast {
    write!(f, "<{:?} as {:?}>::", self_type, trait_)?
} else {
    write!(f, "{:?}::", self_type)?
}
```

The `f.alternate() == true` case is:

```rust
if should_show_cast {
    write!(f, "<{:#} as {:#}>::", self_type, trait_)?
} else {
    write!(f, "{:#}::", self_type)?
}
```